### PR TITLE
fix issue not passing parameters to HTTP integration (#5125)

### PR DESCRIPTION
**Pull request by calvernaz**
Fixes https://github.com/localstack/localstack/issues/5125

Using HTTP_PROXY integration using proxy resources should include all sub-resources and query parameters. This PR adds the query parameters to the resolved uri.

### CODE REVIEW

File: localstack/services/apigateway/apigateway_listener.py [hunk 1]
```diff
@@ -370,7 +373,9 @@ def apply_request_parameters(uri: str, integration: Dict[str, Any], path_params:
             request_param_value = f"method.request.path.{key}"
             if request_parameters.get(request_param_key, None) == request_param_value:
                 uri = uri.replace(f"{{{key}}}", path_params[key])
-    return uri
+
+    # include the request query to the uri
+    return f"{uri}?{urlencode(query_params)}"
```
Comment (whummer) [inline localstack/services/apigateway/apigateway_listener.py:378]: Wondering if we can rely on `uri` not containing a `?`, or whether the integration URI itself may contain query parameters as well. 

Perhaps the logic would be a bit more robust (also against future changes) if we introduce a small utility function like `add_query_params_to_url(..)` in `utils/http_utils.py` that adds query parameters to an existing arbitrary URL (which may or may not contain query params). (We could also add an option whether to overwrite or expand any existing query params in this function.)

**Review by whummer — APPROVED** [review]
Looks great, thanks @calvernaz ! 👍 Only two minor comments, but we can also tackle them in a follow-up iteration, shouldn't hold back the merge..

File: localstack/services/apigateway/apigateway_listener.py (hunk 1 above)
Comment (calvernaz) [reply localstack/services/apigateway/apigateway_listener.py:378]: Thanks, agreed!

File: localstack/services/apigateway/apigateway_listener.py [hunk 2]
```diff
@@ -886,7 +892,9 @@ def invoke_rest_api_integration_backend(
         data = apply_template(integration, "request", data)
         if isinstance(data, dict):
             data = json.dumps(data)
-        uri = apply_request_parameters(uri, integration=integration, path_params=path_params)
+        uri = apply_request_parameters(
```
[the remark this answers is not in the document]
Comment (calvernaz) [reply localstack/services/apigateway/apigateway_listener.py:895]: Gotcha, nice catch! I think I'll include that in this PR. There is a mapping for query parameters, similar to what is done for path parameters.

**Pull request merged**